### PR TITLE
Fix #288, Add Workflow README

### LINKED ISSUE #288: Provide README in .github/workflows
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I reviewed the [cFS README.md file](https://github.com/nasa/cFS/blob/main/README.md) to see if the feature is in the major future work.
* [x] I performed a cursory search to see if the feature request is relevant, not redundant, nor in conflict with other tickets.

**Is your feature request related to a problem? Please describe.**
Users may not know where to find detailed information about the workflows. 

**Describe the solution you'd like**
Provide a description of workflows, how to use them, and how to modify them in a README.md file. Provide a link to this file in the contributing guide. 

**Describe alternatives you've considered**
Do not create a new README.md file and instead add detailed descriptions of the workflows in the contributing guide. 

**Requester Info**
Ariel Adams, ASRC Federal

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate Contributor License agreement to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #288 
Add a README in .github/workflows

**Expected behavior changes**
Users will know what each workflow does.

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+##### CodeQL Analysis
+[![CodeQL Analaysis](https://github.com/nasa/cfs/actions/workflows/codeql-build.yml/badge.svg)](https://github.com/nasa/cfs/actions/workflows/codeql-build.yml)
+
+This action runs GitHub's industry-leading static analysis engine, CodeQL, against our repository's source code to find security vulnerabilities. It then automatically uploads the results to GitHub so they can be displayed in the repository's code scanning alerts found under the security tab. CodeQL runs an extensible set of [queries](https://github.com/github/codeql), which have been developed by the community and the [GitHub Security Lab](https://securitylab.github.com/) to find common vulnerabilities in your code.
+
+Our CodeQL action uses a configuration file to use specific queries, which can be found at [.github/codeql](https://github.com/nasa/cFS/tree/main/.github/codeql).
+
+
+##### Static Analaysis
```
Comment (astrogeco) [inline .github/workflows/README.md:18]: Embed link to cppcheck

File: .github/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+[![Static Analaysis](https://github.com/nasa/cfs/actions/workflows/static-analysis.yml/badge.svg)](https://github.com/nasa/cfs/actions/workflows/static-analysis.yml)
+
+This action runs a static analysis tool for C/C++ code known as cppcheck. Cppcheck is designed to be able to analyze C/C++ code even if it has non-standard syntax, which is common in embedded projects.
+
+##### Deprecated Build, Test, and Run
+[![Deprecated Build, Test, and Run](https://github.com/nasa/cfs/actions/workflows/build-cfs-deprecated.yml/badge.svg)](https://github.com/nasa/cfs/actions/workflows/build-cfs-deprecated.yml)
+
+This action builds, tests, and runs the cFS bundle using deprecated code.
```
Comment (astrogeco) [inline .github/workflows/README.md:26]: mention the OMIT_DEPRECATED flag and maybe link to the cmake documantation

File: .github/workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+##### Build, Test, and Run [OMIT_DEPRECATED=true]
+[![Build, Test, and Run %5B OMIT_DEPRECATED=true %5B](https://github.com/nasa/cfs/actions/workflows/build-cfs.yml/badge.svg)](https://github.com/nasa/cfs/actions/workflows/build-cfs.yml)
+
+This action builds, tests, and runs the cFS bundle omitting deprecated code.
+
+##### Documentation and Guides
+[![Documentation and Guides](https://github.com/nasa/cfs/actions/workflows/build-documentation.yml/badge.svg)](https://github.com/nasa/cfs/actions/workflows/build-documentation.yml)
+
+This action creates doxygen documents for cFE, cFS users guide, and osal guide.
```
Comment (astrogeco) [inline .github/workflows/README.md:36]: maybe add that it also pushes the pdf for pushes to `main`

**Review by astrogeco — CHANGES_REQUESTED** [review]
Looks good, can you fix the headings so that they are second level `##` instead?

File: .github/workflows/README.md (hunk 2 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/README.md]: Where is the cmake documentation?

File: .github/workflows/README.md (hunk 3 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/README.md:47]: Added: 
> Documentation and Guides is ran for every push and every pull request on all branches of cFS in Github Actions. The workflow pushes the PDFs for every push on the main branch.

File: .github/workflows/README.md (hunk 1 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/README.md]: Added
> For more information about Cppcheck, visit http://cppcheck.sourceforge.net/.

File: .github/workflows/README.md (hunk 2 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/README.md]: Linked to global_build_options.cmake

File: .github/workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# Our Workflows
+
+## Format Check
+[![Format Check](https://github.com/nasa/osal/actions/workflows/format-check.yml/badge.svg)](https://github.com/nasa/osal/actions/workflows/format-check.yml)
+
+This action uses clang-format-10 to check for format errors.
```
Comment (astrogeco) [inline .github/workflows/README.md:6]: Add link to .clang-format config file

File: .github/workflows/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+## CodeQL Analysis
+[![CodeQL Analaysis](https://github.com/nasa/cfs/actions/workflows/codeql-build.yml/badge.svg)](https://github.com/nasa/cfs/actions/workflows/codeql-build.yml)
+
+This action runs GitHub's industry-leading static analysis engine, CodeQL, against our repository's source code to find security vulnerabilities. It then automatically uploads the results to GitHub so they can be displayed in the repository's code scanning alerts found under the security tab. CodeQL runs an extensible set of [queries](https://github.com/github/codeql), which have been developed by the community and the [GitHub Security Lab](https://securitylab.github.com/) to find common vulnerabilities in your code.
```
Comment (astrogeco) [inline .github/workflows/README.md:11]: remove the "industry leading" part we can't be seen as "endorsing" a product

File: .github/workflows/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+CodeQL is ran for every push and pull-request on all branches of cFS in GitHub Actions.
```
Comment (astrogeco) [inline .github/workflows/README.md:13]: Use active voice by replacing "is ran" with "runs". Check rest of documentation for this type of usage

File: .github/workflows/README.md (hunk 3 above)
Comment (astrogeco) [reply .github/workflows/README.md:47]: good, we might also want to add the link to where the documentation is pushed to: 
https://github.com/nasa/cfs/tree/gh-pages

**Review by astrogeco — CHANGES_REQUESTED** [review]
Some minor comments. We might want to change the order of these and have the build ones up first. 

Also, replace the section that covers this in Contributing.md with a link to this new readme

File: .github/workflows/README.md (hunk 5 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/README.md]: Removed

File: .github/workflows/README.md (hunk 6 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/README.md]: Changed to "runs" for all "is ran".

File: .github/workflows/README.md (hunk 4 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/README.md]: This link https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormat.html?

Comment (ArielSAdamsNASA) [reply .github/workflows/README.md]: Used https://github.com/nasa/cFS/blob/main/.clang-format

**Pull request merged**